python/tvm/module.py

```diff
@@ -162,13 +162,20 @@ def export_library(self,
                     f.write(_PackImportsToC(self, is_system_lib))
                 files.append(path_cc)
 
-        if has_c_module:
-            options = []
-            if "options" in kwargs:
-                opts = kwargs["options"]
-                options = opts if isinstance(opts, (list, tuple)) else [opts]
-            opts = options + ["-I" + path for path in find_include_path()]
-            kwargs.update({'options': opts})
+        # Make sure we won't pass {'options': None} or {'options': [None, '-std=c++11', ...]}
+        # to compiler. We can not prevent users doing the following code:
+        # f.export_library(path_dso, cc.create_shared, options=None)
+        # f.export_library(path_dso, cc.create_shared, options=[None, '-std=c++11'])
+        if "options" in kwargs or has_c_module:
+            opts = kwargs["options"] if "options" in kwargs else None
+            options = opts if isinstance(opts, (list, tuple)) else [opts]
+            options = [opt for opt in options if opt]
+            if has_c_module:
+                options = options + ["-I" + path for path in find_include_path()]
+            if not options:
+                del kwargs["options"]
+            else:
+                kwargs.update({'options': options})
```

Comment:
I feel like we don't have to test `has_c_module` twice. Maybe the following logic is sufficient?
```
options = ["-I" +path for path in find_include_path()] if has_c_module else []
if "options" in kwargs:
  opts = kwargs["options"]
  options += opts if isinstance(opts, (list, tuple)) else [opts]
  options = [opt for opt in options if opt]
  if not options:
    del kwargs["options"]
if options:
  kwargs.update({"options": options})
```

Reply:
Yes. Your way could avoid twice check of `has_c_module`.

Reply:
Done. Please review it again.